src/yarp.c

```diff
@@ -2075,7 +2053,46 @@ parse_arguments(yp_parser_t *parser, yp_node_t *arguments, yp_token_type_t termi
       expect(parser, YP_TOKEN_COMMA, "Expected a ',' to delimit arguments.");
     }
 
-    yp_node_t *argument = parse_argument(parser, arguments);
+    if (parsed_block_argument) {
+      yp_error_list_append(&parser->error_list, "Unexpected argument after block argument.", parser->current.start - parser->start);
+    }
+
+    yp_node_t *argument;
+
+    if (accept(parser, YP_TOKEN_STAR)) {
+      yp_token_t previous = parser->previous;
+      if (parser->current.type == YP_TOKEN_PARENTHESIS_RIGHT || parser->current.type == YP_TOKEN_COMMA) {
+        if (!yp_token_list_includes(&parser->current_scope->as.scope.locals, &parser->previous)) {
+          yp_error_list_append(&parser->error_list, "unexpected * when parent method is not forwarding.", parser->previous.start - parser->start);
+        }
+        argument = yp_node_star_node_create(parser, &previous, NULL);
+      } else {
+        if (parsed_double_splat_argument) {
+          yp_error_list_append(&parser->error_list, "Unexpected splat argument after double splat.", parser->current.start - parser->start);
+        }
+        yp_token_t operator = parser->previous;
+        yp_node_t *value = parse_expression(parser, BINDING_POWER_DEFINED, "Expected to be able to parse an argument.");
```

Comment:
Using `BINDING_POWER_DEFINED` like it was in the `parse_argument` which was merged here to make working with flags easier.